### PR TITLE
allow org members to view circle members pages

### PR DESCRIPTION
**Pull request by levity**
This PR makes the circle members page accessible to org members. This means:

- As an org member (not a circle member), you can mouse over circles in the org page and see a link to the Members page, which you can click on
- When you do, the Members page will be loaded, and the side nav will show the circle you've selected (which doesn't show otherwise, since you're not a member
- The Members page has limited content: there's no admin button or Leave Circle button, of course

### Testing

Since there is no way in the UI to add a member directly to an org yet, you can do this from the REPL: run `await createMembersFromCircles(orgId)`, where `orgId` is the org you want to add members to. All circle members in that org will be made org members as well.

If you want to scrutinize the permissions setup, you can remove yourself from all circles in the org, and you should still be able to see each circle's members page in the limited way described above.

If you want to test this in the preview app, you need to also deploy this code to staging, and then run the REPL against staging (by setting NODE_HASURA_URL and HASURA_GRAPHQL_ADMIN_SECRET to staging values)

### Todo

- [x] stop using useSelectedCircle
- [x] combine queries on members page
- [x] set up permissions so org members can view circle, users, profiles
- [x] move MemberRow to its own file
- [x] select correct org in side nav when viewing a circle you're not part of
- [x] show non-member circles in side nav only when selected
- [x] fix circle overview page (deferred to future PR)
- [x] get map page working (deferred to future PR)
- [x] debug failing assertion when switching from non-member page to give page for a circle you're in
  - root cause: useSelectedCircle called before it's updated for the new circle

### CODE REVIEW

**Comment by levity** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1993** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1993" target="_blank">Graphite</a> 
    * **PR #1994** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1994" target="_blank">Graphite</a> 
      * **PR #1995** <a href="https://app.graphite.dev/github/pr/coordinape/coordinape/1995" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1995?utm_source=stack-comment).

File: src/hooks/migration.ts [hunk 1]
```diff
@@ -27,20 +27,27 @@ export const useFixCircleState = (circleId: number | undefined) => {
...
       logger.log(`circle ids match`);
+      setReady(true);
     } else if (fullCircles.has(circleId)) {
       logger.log(`reusing circle data: ${circleId}`);
       setCircleIdSource(circleId);
+      setReady(true);
     } else {
       logger.log(`fetching circle data: ${circleId}`);
-      fetchCircle({ circleId, select: true });
+      fetchCircle({ circleId, select: true }).then(() => setReady(true));
```
Comment (levity) [inline src/hooks/migration.ts:46]: this prevents a race condition where a new circle's page starts rendering before the data for it is ready, i.e. it uses the previous circle's data

File: src/hooks/useConnectedAddress.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
+import assert from 'assert';
 import { useEffect, useState } from 'react';
 
 import { useWeb3React } from 'hooks/useWeb3React';
 
-export default function useConnectedAddress() {
+function useConnectedAddress(required: true): string;
```
Comment (levity) [inline src/hooks/useConnectedAddress.ts:6]: https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

File: src/recoilState/app.ts [hunk 3]
```diff
@@ -123,23 +123,13 @@ export const rCircle = selectorFamily<ICircleState, number | undefined>({
...
-
-      const me = { ...myUser, profile: myProfile };
+      const me = myUser ? { ...myUser, profile: myProfile } : undefined;
 
       return {
         circleId,
         circle,
         myUser: me,
         users: getCircleUsers().toArray(),
-        usersNotMe: getCircleUsers()
```
Comment (levity) [inline src/recoilState/app.ts:137]: this was already unused

File: src/pages/MembersPage/MembersTable.tsx [hunk 4]
```diff
@@ -931,38 +933,45 @@ export const MembersTable = ({
...
+            token_name={circle.token_name}
             myUser={me}
             setDeleteUserDialog={setDeleteUserDialog}
-            setLeaveCircleDialog={setLeaveCircleDialog}
+            showLeaveModal={() => setShowLeave(true)}
             circleId={circle.id}
           />
         )}
       </MemberTable>
+      <LeaveCircleModal
```
Comment (levity) [inline src/pages/MembersPage/MembersTable.tsx:968]: moving this modal from MembersPage/index wasn't strictly necessary for this work; it just seemed opportune to refactor to reduce the amount of prop-drilling

File: src/pages/MembersPage/MembersTable.tsx [hunk 5]
```diff
@@ -187,29 +184,28 @@ const UserName = ({ user }: { user: ICircleUser }) => {
   );
 };
 
-const MemberRow = ({
+export const MemberRow = ({
```
Comment (levity) [inline src/pages/MembersPage/MembersTable.tsx:187]: i originally pulled this out into its own file but it made the diff much bigger

File: src/hooks/migration.ts (hunk 1 above)
Comment (crabsinger) [reply src/hooks/migration.ts:46]: Not sure this really would work with all this recoil stuff, but RQ exposes an isFetching prop that maybe could serve as a similar purpose to hold off on page load until the query is complete?

**Comment by crabsinger** [comment]
Adding a note to this PR in case you didn't see this: https://github.com/coordinape/coordinape/pull/1944/files#r1131506108

**Review by crabsinger — COMMENTED** [review]
UI:

I added my user to the org, but not the circle: I see the Activity, Map, and Members page show on the nav bar, but Activity and Map are not yet supported so they just redirect to /circles.

Feels a bit weird to ship with those links broken, maybe they're disabled or removed for now until those pages support org members viewing?
Screenshot 2023-03-09 at 11 42 20 AM

File: src/features/nav/getNavData.ts [hunk 6]
```diff
@@ -15,14 +15,7 @@ export const getNavData = (address: string, chainId: number) =>
...
           circles: [
-            {
-              where: {
-                users: {
-                  address: { _eq: address.toLowerCase() },
-                  deleted_at: { _is_null: true },
-                },
-              },
-            },
+            {},
```
[the remark this answers is not in the document]
Comment (crabsinger) [reply src/features/nav/getNavData.ts:18]: maybe the default permissions handle this already for us?

File: src/pages/MembersPage/getMembersPageData.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
+import { order_by } from 'lib/gql/__generated__/zeus';
+import { client } from 'lib/gql/client';
+
+import { Awaited } from 'types/shim';
+
+export const getMembersPageData = async (circleId: number) => {
```
Comment (crabsinger) [inline src/pages/MembersPage/getMembersPageData.ts:6]: maybe we should rename this to `CircleMembers` or `OrgMembers` now to differentiate these two...

**Review by crabsinger — COMMENTED** [review]
a few comments and questions.

**Comment by levity** [comment]
> UI:
> 
> 
> 
> I added my user to the org, but not the circle: I see the Activity, Map, and Members page show on the nav bar, but Activity and Map are not yet supported so they just redirect to /circles.
> 
> 
> 
> Feels a bit weird to ship with those links broken, maybe they're disabled or removed for now until those pages support org members viewing?
> 
> Screenshot 2023-03-09 at 11 42 20 AM
> 
> 

Hmm, the map link showing is a bug, I thought I removed that. I didn't check activity because I don't have it enabled locally

File: src/features/nav/getNavData.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (levity) [reply src/features/nav/getNavData.ts:18]: yep, soft-deleted circles are excluded. this did prompt me to notice that we don't do the same for _users_ of soft-deleted circles tho; spun that out into #2023

[the remark this answers is not in the document]
Comment (levity) [reply src/features/nav/getNavData.ts:18]: i'll also make this change for the org_members table itself

File: src/pages/MembersPage/getMembersPageData.ts (hunk 7 above)
Comment (levity) [reply src/pages/MembersPage/getMembersPageData.ts:6]: I'll do this renaming in a later PR

File: hasura/metadata/databases/default/tables/public_users.yaml [hunk 8]
```diff
@@ -148,13 +148,25 @@ select_permissions:
...
-                    profile:
-                      id:
-                        _eq: X-Hasura-User-Id
+              deleted_at:
+                _is_null: true
+          - _or:
+              - circle:
+                  users:
+                    _and:
+                      - deleted_at:
```
Comment (crabsinger) [inline hasura/metadata/databases/default/tables/public_users.yaml:157]: a bit pedantic,but I've been putting the `deleted_at` clause as second after the profile check consistently for easier scanning and readability

**Review by crabsinger — DISMISSED** [review]
looks good!

**Comment by levity** [comment]
@levity queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1995).

**Comment by levity** [comment]
@levity merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coordinape/coordinape/1995).

**Pull request merged**